qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,708 @@
+            base_fitter: A callback function to perform fitting with formatted data.
+            data_processor: A callback function to format experiment data.
+            x_key: Circuit metadata key representing a scanned value.
+            plot: Set ``True`` to create figure for fit result.
+            ax: Optional. A matplotlib axis object to draw.
+            xlabel: X label of fit result figure.
+            ylabel: Y label of fit result figure.
+            fit_reports: Mapping of fit parameters and representation in the fit report.
+        """
+        return Options(
+            base_fitter=multi_curve_fit,
+            data_processor=level2_probability,
+            x_key="xval",
+            plot=True,
+            ax=None,
+            xlabel="x value",
+            ylabel="y value",
+            fit_reports=None,
+        )
+
+    def _create_figures(
+        self,
+        x_values: np.ndarray,
+        y_values: np.ndarray,
+        y_sigmas: np.ndarray,
+        series: np.ndarray,
+        analysis_results: AnalysisResult,
+        axis: Optional["AxisSubplot"] = None,
+        xlabel: str = "x value",
+        ylabel: str = "y value",
+        fit_reports: Optional[Dict[str, str]] = None,
+    ) -> List["Figure"]:
+        """Create new figures with the fit result and raw data.
+
+        Subclass can override this method to create different type of figures.
+
+        Args:
+            x_values: Full data set of x values.
+            y_values: Full data set of y values.
+            y_sigmas: Full data set of y sigmas.
+            series: An integer array representing a mapping of data location to series index.
+            analysis_results: Analysis result containing fit parameters.
+            axis: User provided axis to draw result.
+            xlabel: String shown in figure x axis label.
+            ylabel: String shown in figure y axis label.
+            fit_reports: Mapping of fit parameters and representation in the fit report.
+
+        Returns:
+            List of figures.
+        """
+        if plotting.HAS_MATPLOTLIB:
+
+            if axis is None:
+                figure = plotting.pyplot.figure(figsize=(8, 5))
+                axis = figure.subplots(nrows=1, ncols=1)
+            else:
+                figure = axis.get_figure()
+
+            axis.set_xlabel(xlabel, fontsize=16)
+            axis.set_ylabel(ylabel, fontsize=16)
+
+            for series_def in self.__series__:
+
+                # plot raw data
+
+                xdata, ydata, _ = self._subset_data(
+                    series_def.name, x_values, y_values, y_sigmas, series
+                )
+                plotting.plot_scatter(xdata=xdata, ydata=ydata, ax=axis, zorder=0)
+
+                # plot formatted data
+
+                xdata, ydata, sigma = self._subset_data(
+                    series_def.name, *self._pre_processing(x_values, y_values, y_sigmas, series)
+                )
+                plotting.plot_errorbar(
```

Comment:
We can probably remove the dotted line in the error bar plot and only plot the points